### PR TITLE
Support for CentOS stream 9

### PR DESCRIPTION
**Pull request by yairpod**
Adding support for CentOS Stream 9 as SEAPATH Hosts.

This is meant to share code with the Debian variation.
For CentOS Stream installations you can use my reference Kickstart file:
https://github.com/yairpod/SEAPATH_CentOS_Stream_Kickstart

API changes:
- changes inventory variable skip_reboot_setup_debian renaming it to skip_reboot_setup
- Inventroy variable extra_debiansnmp_sudoers is changed and renamed extra_usersnmp_sudoers

### CODE REVIEW

File: playbooks/test_run_cukinia.yaml [hunk 1]
```diff
@@ -24,7 +29,7 @@
       shell:
         cmd: >-
           MACHINENAME={{ cukinia_namespace | default(inventory_hostname) }}
-          cukinia -f {{ tests_format }}
+          {{ path_command_path }} -f {{ tests_format }}
```
[the remark this answers is not in the document]
Comment (yairpod) [reply playbooks/test_run_cukinia.yaml]: Thank, Fixed in 1d88f95

File: playbooks/test_run_cukinia.yaml [hunk 2]
```diff
@@ -35,16 +40,21 @@
...
   tasks:
+    - include_vars: ../vars/debian_paths.yml
+      when: ansible_distribution == 'Debian'
+    - include_vars: ../vars/centos_paths.yml
+      when: ansible_distribution == 'CentOS'
     - name: Run tests cluster test
       shell:
         cmd: >-
-          cukinia -f {{ tests_format }}
+          {{ path_command_path }} -f {{ tests_format }}
```
[the remark this answers is not in the document]
Comment (yairpod) [reply playbooks/test_run_cukinia.yaml]: Thank, Fixed in 1d88f95

**Comment by ebail** [comment]
Hi @yairpod. Thanks for your pull request.

So I guess that we would need to add https://github.com/yairpod/SEAPATH_CentOS_Stream_Kickstart as a repo in LFEnergy SEAPATH.

As you know, we have an intensive CI approach. I would like to have a CI for this SEAPATH flavor so that we can monitor the all the tests succeed. 
Do you think you could provide a running CI @yairpod ?
By the way did you check that all the tests are green with your distro ?

Also we should consider an other name for "debian-main". We would have 3 different OS of SEAPATH: Yocto, Debian and centos.
The best would be to merge all those flavors in a common Ansible branch. As a first step, we would need to rename debian-main into precompiled-main for instance.

@insatomcat @dupremathieu what is your point of view ?

Best,

Eloi

**Comment by insatomcat** [comment]
I'd prefer "debiancentos" over "precompiled".
And yes, this is great work but we would need some form of engagement that there would be someone to maintain centos compatibility long term, including maintaining the infrastructure to run CI testing.
Seapath is quite a dynamic project, it will require work to make sure all changes work on centos and this is something noone on the current team has time for. Is it something you or Redhat is ready to do?
If yes I think we need to discuss this during next TSC.

File: playbooks/cluster_setup_debian_centos.yaml [hunk 3]
```diff
@@ -0,0 +1,71 @@
+# This playbook setup SEAPATH on both debian and CentOS machines
+# The machine must previously have been flashed
+# with an iso created by build_debian_iso or CentOS with appropriate kickstart file
+# This playbooks can also configure VMs created with build_debian_iso
+
+- name: Check if we are using a correct seapath flavor
+  hosts:
+    - cluster_machines
+    - standalone_machine
+    - VMs
```
Comment (dupremathieu) [inline playbooks/cluster_setup_debian_centos.yaml:10]: VMs should not to be here.

File: playbooks/cluster_setup_debian_centos.yaml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  tasks:
+    - name: Retrieve redhat-release
```
Comment (dupremathieu) [inline playbooks/cluster_setup_debian_centos.yaml:12]: Ansible facts and `ansible_distribution`

File: playbooks/cluster_setup_debian_centos.yaml [hunk 5]
```diff
@@ -0,0 +1,71 @@
+# This playbook setup SEAPATH on both debian and CentOS machines
```
Comment (dupremathieu) [inline playbooks/cluster_setup_debian_centos.yaml:1]: I preferred naming it `cluster_setup.yaml` or `cluster_setup_main.yaml` and merging this file with cluster_setup_debian.yaml and cluster_setup_centos.yaml.

File: playbooks/cluster_setup_ha.yaml [hunk 6]
```diff
@@ -5,6 +5,15 @@
 # Pacemaker.
 
 ---
+
+- name: Get distrebution variables
+  hosts:
+    - cluster_machines
+    - standalone_machine
+    - VMs
```
Comment (dupremathieu) [inline playbooks/cluster_setup_ha.yaml:13]: VMs should not to be here.

File: playbooks/cluster_setup_ha.yaml [hunk 7]
```diff
@@ -100,6 +109,7 @@
       when: groups['valid_machine'] is defined
 
 - name: Setup Pacemaker
+  gather_facts: True
```
Comment (dupremathieu) [inline playbooks/cluster_setup_ha.yaml:112]: True -> true

File: playbooks/cluster_setup_network.yaml [hunk 8]
```diff
@@ -172,6 +180,7 @@
       when: cluster_protocol is defined and cluster_protocol == "HSR" and hsr_mac_address is defined
 
 - name: Configure OVS
+  gather_facts: True
```
Comment (dupremathieu) [inline playbooks/cluster_setup_network.yaml:183]: True -> true :)

File: playbooks/cluster_setup_network.yaml [hunk 9]
```diff
@@ -364,6 +373,7 @@
     when: dns_servers is not defined
 
 - name: Configure TimeMaster
+  gather_facts: True
```
Comment (dupremathieu) [inline playbooks/cluster_setup_network.yaml:376]: True -> true

File: playbooks/cluster_setup_network.yaml [hunk 10]
```diff
@@ -417,6 +427,7 @@
         - apply_config or need_reboot is not defined or not need_reboot
 
 - name: Stop chrony service
+  gather_facts: True
```
Comment (dupremathieu) [inline playbooks/cluster_setup_network.yaml:430]: True -> true

File: playbooks/cluster_setup_prerequiscentos.yaml [hunk 11]
```diff
@@ -0,0 +1,80 @@
+# Copyright (C) 2023, RTE (http://www.rte-france.com)
```
Comment (dupremathieu) [inline playbooks/cluster_setup_prerequiscentos.yaml:1]: Copyright

File: playbooks/test_run_cukinia.yaml [hunk 12]
```diff
@@ -5,11 +5,21 @@
...
+    - VMs
+  tasks:
+  - include_vars: "../vars/{{ ansible_distribution }}_paths.yml"
+
 - hosts:
     - cluster_machines
     - standalone_machine
     - VMs
   name: Cukinia tests
+  gather_facts: True
```
Comment (dupremathieu) [inline playbooks/test_run_cukinia.yaml:22]: True -> true

File: playbooks/test_run_cukinia.yaml [hunk 13]
```diff
@@ -35,6 +45,7 @@
 - hosts:
     - cluster_machines
   name: Cukinia tests
+  gather_facts: True
```
Comment (dupremathieu) [inline playbooks/test_run_cukinia.yaml:48]: True -> true

File: roles/centos/handlers/main.yml [hunk 14]
```diff
@@ -0,0 +1,8 @@
+- name: daemon-reload
+  ansible.builtin.service:
+    daemon_reload: yes
+
+- name: restart syslog-ng
+  ansible.builtin.systemd:
+    name: syslog-ng
+    state: restarted
```
Comment (dupremathieu) [inline roles/centos/handlers/main.yml:8]: Add finale new line at the end of the file.

File: roles/centos/handlers/main.yml [hunk 15]
```diff
@@ -0,0 +1,8 @@
+- name: daemon-reload
```
Comment (dupremathieu) [inline roles/centos/handlers/main.yml:1]: Header

File: roles/centos/hypervisor/tasks/main.yml [hunk 16]
```diff
@@ -0,0 +1,320 @@
+- name: enable docker.service
```
Comment (dupremathieu) [inline roles/centos/hypervisor/tasks/main.yml:1]: Header

File: roles/centos/physical_machine/handlers/main.yml [hunk 17]
```diff
@@ -0,0 +1,3 @@
+- name: daemon-reload
```
Comment (dupremathieu) [inline roles/centos/physical_machine/handlers/main.yml:1]: Header

File: roles/centos/physical_machine/tasks/main.yml [hunk 18]
```diff
@@ -0,0 +1,461 @@
+# We need to restart nginx if it's running and know if gunicorn socket is present
```
Comment (dupremathieu) [inline roles/centos/physical_machine/tasks/main.yml:1]: Header

File: roles/centos/tasks/main.yml [hunk 19]
```diff
@@ -0,0 +1,199 @@
+- name: disable vim defaults
```
Comment (dupremathieu) [inline roles/centos/tasks/main.yml:1]: Header

File: src/centos/seapath-config_ovs.service [hunk 20]
```diff
@@ -0,0 +1,17 @@
+# Copyright (C) 2021, RTE (http://www.rte-france.com)
```
Comment (dupremathieu) [inline src/centos/seapath-config_ovs.service:1]: Update header

File: src/centos/taskset_boot.sh.j2 [hunk 21]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
```
Comment (dupremathieu) [inline src/centos/taskset_boot.sh.j2:2]: Header

File: src/centos/taskset_boot.sh.j2 [hunk 22]
```diff
@@ -0,0 +1,6 @@
...
+/usr/bin/taskset -acp {{ cpusystem }} 2
+/usr/bin/taskset -acp {{ cpusystem }} $(pgrep rcu_preempt)
```
Comment (dupremathieu) [inline src/centos/taskset_boot.sh.j2:4]: I think it is already done by tunectl, isn't?

File: src/debian/modules/netfilter.conf [hunk 23]
```diff
@@ -0,0 +1,2 @@
+br_netfilter
```
Comment (dupremathieu) [inline src/debian/modules/netfilter.conf:1]: Unrelated to Centos.

File: src/debian/modules/vhost_vsock.conf [hunk 24]
```diff
@@ -0,0 +1 @@
+vhost_vsock
```
Comment (dupremathieu) [inline src/debian/modules/vhost_vsock.conf:1]: Unrelated to Centos.

File: src/debian/sriov_driver.conf [hunk 25]
```diff
@@ -0,0 +1 @@
+sriov_driver
```
Comment (dupremathieu) [inline src/debian/sriov_driver.conf:1]: Unrelated to Centos.

File: roles/centos/hypervisor/tasks/main.yml [hunk 26]
```diff
@@ -0,0 +1,320 @@
...
+  when: sriov_driver is defined
+- name: load the sriov module
+  community.general.modprobe:
+    name: "{{ sriov_driver }}"
+    state: present
+  when: sriov_driver is defined
+
+- name: add sriov sysfs rules
+  template:
+    src: ../templates/sriov.conf.j2
```
Comment (dupremathieu) [inline roles/centos/hypervisor/tasks/main.yml:36]: Move it inside role templates directory

File: templates/sriov.conf.j2 [hunk 27]
```diff
@@ -0,0 +1,2 @@
+#Type  Path                                                   Mode User Group Age Argument…
```
Comment (dupremathieu) [inline templates/sriov.conf.j2:1]: Move it inside role templates directory

File: templates/tmpfiles-workqueue_cpumask.conf.j2 [hunk 28]
```diff
@@ -0,0 +1,6 @@
+#Type  Path                                                   Mode User Group Age Argument…
```
Comment (dupremathieu) [inline templates/tmpfiles-workqueue_cpumask.conf.j2:1]: Move it inside role templates directory

File: src/centos/sudoers/Centos-snmp.j2 [hunk 29]
```diff
@@ -0,0 +1,4 @@
+Defaults:Debian-snmp !requiretty
```
Comment (dupremathieu) [inline src/centos/sudoers/Centos-snmp.j2:1]: {{ snmp_user_name }}

File: src/centos/sudoers/Centos-snmp.j2 [hunk 30]
```diff
@@ -0,0 +1,4 @@
...
+{% if extra_debiansnmp_sudoers is defined %}
```
Comment (dupremathieu) [inline src/centos/sudoers/Centos-snmp.j2:2]: change variable name

**Review by dupremathieu — CHANGES_REQUESTED** [review]
Greats
Sorry for the number of comments, it is mostly cosmetic issues.

File: playbooks/cluster_setup_centos.yaml [hunk 31]
```diff
@@ -0,0 +1,57 @@
+# Copyright (C) 2023, RTE (http://www.rte-france.com)
```
[the remark this answers is not in the document]
Comment (yairpod) [reply playbooks/cluster_setup_centos.yaml:1]: One issue is that during testing the cluster_setup_debian_centos (that really needs renaming) takes significantly longer to run as it does go through all Debian and CentOS setup steps, So I would keep the dedicated OS specific setup playbooks.

File: playbooks/cluster_setup_debian_centos.yaml (hunk 5 above)
Comment (yairpod) [reply playbooks/cluster_setup_debian_centos.yaml:1]: I would keep the OS specific playbooks (as written in another comment),
But this playboooks should be renamed, Would cluster_setup_all.yaml be a good name?

File: playbooks/cluster_setup_centos.yaml [hunk 32]
```diff
@@ -0,0 +1,57 @@
...
+  tasks:
+    - name: Retrieve Seapath system version
+      command: "cat /etc/redhat-release"
+      register: "uname_output"
+
+    - name: Check if current Ansible branch is compatible
+      fail:
+        msg: "The current branch (debian-main) is incompatible with the Yocto version of SEAPATH. Please use the main branch to configure this machine"
+      failed_when:
+        - not(uname_output.stdout | regex_search("CentOS"))
```
[the remark this answers is not in the document]
Comment (yairpod) [reply playbooks/cluster_setup_centos.yaml]: Added :)

File: roles/centos/hypervisor/tasks/main.yml (hunk 26 above)
Comment (yairpod) [reply roles/centos/hypervisor/tasks/main.yml]: Moved

File: playbooks/cluster_setup_debian_centos.yaml (hunk 4 above)
Comment (yairpod) [reply playbooks/cluster_setup_debian_centos.yaml]: Fixed

File: playbooks/cluster_setup_debian_centos.yaml (hunk 5 above)
Comment (dupremathieu) [reply playbooks/cluster_setup_debian_centos.yaml:4]: Ok

**Review by dupremathieu — CHANGES_REQUESTED** [review]
There are some unresolved conversation.

File: src/centos/taskset_boot.sh.j2 (hunk 22 above)
Comment (yairpod) [reply src/centos/taskset_boot.sh.j2]: removed

File: src/centos/sudoers/Centos-snmp.j2 (hunk 30 above)
Comment (yairpod) [reply src/centos/sudoers/Centos-snmp.j2]: Changed

**Review by dupremathieu — APPROVED** [review]

**Pull request merged**